sql/parser/eval.go

```diff
@@ -259,6 +263,147 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// DDate is the date Datum.
+type DDate struct {
+	time.Time
+}
+
+// Type implements the Datum interface.
+func (d DDate) Type() string {
+	return "date"
+}
+
+// Compare implements the Datum interface.
+func (d DDate) Compare(other Datum) int {
+	if other == DNull {
+		// NULL is less than any non-NULL value.
+		return 1
+	}
+	v, ok := other.(DDate)
+	if !ok {
+		panic(fmt.Sprintf("unsupported comparison: %s to %s", d.Type(), other.Type()))
+	}
+	if d.Before(v.Time) {
+		return -1
+	}
+	if v.Before(d.Time) {
+		return 1
+	}
+	return 0
+}
+
+// Next implements the Datum interface.
+func (d DDate) Next() Datum {
+	return DDate{Time: d.Time.Add(24 * time.Hour)}
+}
+
+// IsMax implements the Datum interface.
+func (d DDate) IsMax() bool {
+	return false
```

Comment:
Add a comment to the `{DDate,DTimestamp}.Is{Max,Min}` functions that dates/timestamps do not have a min or max value.